include/common/common/window_base.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include <concepts>
+
+namespace wnd
+{
+
+using UniqueSurface = vk::UniqueSurfaceKHR;
+
+template <typename Derived> class WindowBase
+{
+  protected:
+    WindowBase() = default;
+
+  public:
+    Derived& impl() { return static_cast<Derived&>( *this ); }
+    const Derived& impl() const { return static_cast<const Derived&>( *this ); }
+
+  public:
+    UniqueSurface createSurface( vk::Instance instance ) const { return impl().createSurface( instance ); }
```

Comment:
Well, it indeed has to create a surface and pass it's ownership to the callee. A raw vk::Surface won't be destroyed correctly by its destructor, because it's a non-owning wrapper. For correctness a RAII UniqueSurface is preferable